### PR TITLE
fix: semantic versions are not calculated correctly

### PR DESCRIPTION
**Pull request by 3h4x**
# Summary of changes

**Summarize** the description of your changes. 

Regex pattern never matched major or minor release. 
The definition in GHA requires leading and trailing `/`

https://github.com/PaulHatch/semantic-version/blob/f9d3daa396c5ee4d2286963a921d0a3f687853ec/src/providers/DefaultVersionClassifier.ts#L22-L35

Additionally I added optional word in parentheses as it's common patter to use. 

Also I added action printing next version, it's not necessary but might be useful in the beginning.

~Interestingly enough it bumps to 0.4.0 where there is no feat commit on this feature branch. It looks like a bug in this action as it sees such commits and run regexs on them:~

## Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue):
    - [ ] Have you **written tests **to protect against future occurrences of the bug?
- [ ] New feature (non-breaking change which adds functionality)
    - [ ] Have you **seen it working** in a development environment?
    - [ ] Have you **written tests** for **happy** and **unhappy** paths?
    - [ ] Have you implemented this feature as per the **issue acceptance criteria**?
- [ ] Breaking change (a feature that would cause existing functionality not to work as expected
    - [ ] Is the breaking change **documented**?
    - [ ] Has any previous changes been **deprecated**?
- [ ] CI/CD
    - [ ] If introducing new actions, have you **looked at the action code** for anything spurious?
    - [ ] Have you written **custom scripts** for things that could be actions already on the marketplace?
    - [ ] If introducing new actions, have you **pegged a static version**?
- [ ] Documentation Update
    - [ ] Have you checked other documentation, such as the docs repository?
    - [ ] If updating script documentation, have you **tested it on both environments**?
- [ ] Substrate
    - [ ] RPC methods?
    - [ ] Chainspec, both JSON specs & the module?
    - [ ] Runtime versioning?
    - [ ] Benchmarks?
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Feel free to explain in further detail your choices if this is a significant change.

### CODE REVIEW

**Comment by AwesomeIbex** [comment]
Hmm, I can't get any matches for this on regextester

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -23,10 +23,8 @@ jobs:
         uses: paulhatch/semantic-version@v5.0.2
         with:
           tag_prefix: "v"
-          major_pattern: "/feat!:|refactor!:"
-          major_regexp_flags: "g"
```
Comment (AwesomeIbex) [inline .github/workflows/release.yml:27]: these need to stay too

Comment (3h4x) [reply .github/workflows/release.yml:27]: you mean the global regex flag?

**Comment by 3h4x** [comment]
> > Anyways, I dont want to spend much more time on it as I thought it will be just a couple of minutes during weekend and spend a couple hours debugging grin If you want you can close this.
> 
> I did test various escape sequences and couldn't get any of them passing. No need to close, just will check this when I get some spare time.

If you run a node and do create regex like in GHA (link in desc) you will get this regexp
```
> RegExp('/^((feat)|(refactor))(\\([a-z]+\\))?!:/'.slice(1, -1))
/^((feat)|(refactor))(\([a-z]+\))?!:/
```

Not sure right now why I used this kind of grouping, more reasonable would be:
```
^(feat|refactor)\(([a-z]+)\)?!:
```
Either way I think only first grouping is used so doesnt matter much at this point.

**Comment by 3h4x** [comment]
Now I understand semantic calculation better and next version is v0.4.0 because it does look at all commits until previous tag.
Looks like tag v0.3.7 was removed and it contained `feat:` commit.

**Pull request merged**